src/main/java/dev/snowdrop/release/model/Release.java

```diff
@@ -99,10 +112,31 @@ public void setTest(boolean test) {
             });
         }
     }
-    
-    public List<String> validate(boolean skipProductRequests) {
+
+    public void setSchedule(String releaseDate, String eolDate) throws ParseException {
+        this.schedule = generateSchedule(releaseDate, eolDate);
+    }
+
+    public Schedule generateSchedule(String releaseDate, String eolDate) throws ParseException {
+        Schedule schedule = new Schedule();
```

Comment:
I'd rather we set the dates in the constructor than add setters as I don't think we'd need (or even allow) a schedule to be modified after it's been created. Immutability FTW!